docs/DEVELOPMENT-GUIDE.md

```diff
@@ -347,10 +347,11 @@ The `high_water_mark` tracks the furthermost byte which has been written but not
 Note that this may be past the `write_cursor` if `s2n_stuffer_rewrite()` has been called.
 Explicitly tracking the `high_water_mark` allows us to track the bytes which need to be wiped, and helps avoids needless zeroing of memory.
 The next two bits of state track whether a stuffer was dynamically allocated (and so should be free'd later) and whether or not it is growable.
-`tainted` is set to 1 whenever the raw access functions are called.
-If a stuffer is currently tainted then it can not be resized and it becomes ungrowable.
+`tainted` is set to true whenever `s2n_stuffer_raw_read/write()` are called.
+If a stuffer is currently tainted then it can not be resized and it becomes ungrowable due to memory safety constraints mentioned above.
```

Comment:
```suggestion
`tainted` is set to true whenever `s2n_stuffer_raw_read/write()` are called.
If a stuffer is currently tainted then it can not be resized and it becomes ungrowable due to memory safety constraints.
```